src/main/java/com/ocient/jdbc/XGConnection.java

```diff
@@ -2195,35 +2299,47 @@ private void clientHandshakeSSO(final String db, final OpenIDAuthenticator authe
 			}
 
 			// Now send the tokens to the database for validation
-			clientHandshakeGCM(username, password, db, shouldRequestVersion, HandshakeType.SSO);
+			SSOHandshakeCompletionStage = clientHandshakeGCM(username, password, db, shouldRequestVersion, HandshakeType.SSO)
+											.thenApply(successOrEx -> {
+												LOGGER.log(Level.INFO, () -> "Handshake SSO without token finished");
+												return successOrEx;
+											});
```

Comment:
I think we should remove the error handling and the fetch server version stuff to DRY this up.
```suggestion
			return clientHandshakeGCM(username, password, db, shouldRequestVersion, HandshakeType.SSO);
```

We can move the log message to the common code path in `clientHAndshake()` and include the handshake type in the message.